### PR TITLE
Externalize config

### PR DESCRIPTION
**Pull request by Dan-M**
### Description
The main intent of this PR is to get rid of the API token (and account id) in the code. This should not be in the source of a public repository.
It introduces a small configuration library to externalise the libraries configuration. Library consumers can provide their own configuration using an application.conf file.

Note that only one of the 3 tokens that have been hard coded in the code can be configured in one VM. We could generally support multiple tokens. 
Based on the current code I didn't see a need for that and left it out.

@mnembrini If you think these changes are Ok and can be merged - could you please revoke current tokens and create a new one.
The new token needs to be configured as Project env variable on circle-ci.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,12 +5,31 @@
 
 harvest-client is a Java client for the Harvest REST API (https://help.getharvest.com/api-v2/)
 
+## Getting started with Development
```
Comment (mnembrini) [inline README.md]: I prefer to have this information in the wiki, I will move it there

File: src/test/java/ch/aaap/harvestclient/examples/UsersApiExample.java [hunk 2]
```diff
@@ -6,16 +6,17 @@
...
 import ch.aaap.harvestclient.domain.User;
 import ch.aaap.harvestclient.domain.param.UserCreationInfo;
-import ch.aaap.harvestclient.util.TestSetupUtil;
 
 public class UsersApiExample {
 
     private static final Logger log = LoggerFactory.getLogger(UsersApiExample.class);
 
-    private Harvest harvest = TestSetupUtil.getAdminClient();
+    private Harvest harvest = new Harvest(ConfigFactory.load());
```
Comment (mnembrini) [inline src/test/java/ch/aaap/harvestclient/examples/UsersApiExample.java]: any reason for not using new Harvest() instead?

**Comment by mnembrini** [comment]
I agree with not having any API tokens in the code, thank you for showing me this nice configuration library. Please note that the API tokens that were hardcoded were all for a test account with no data in it.

File: README.md (hunk 1 above)
Comment (Dan-M) [reply README.md]: I like to have README's that get me up and running without going to the Wiki. The Wiki is ok for further information IMO.
If you look at "README best practices", they all have a very similar structure and one thing they have in common is, that they show you how to get started with development.

But there is no reason not have that in the WIKI as well with more detailed information.

File: src/test/java/ch/aaap/harvestclient/examples/UsersApiExample.java (hunk 2 above)
Comment (Dan-M) [reply src/test/java/ch/aaap/harvestclient/examples/UsersApiExample.java]: Yes. This way the example shows that it is possible to provide your own configuration.

File: README.md (hunk 1 above)
Comment (Dan-M) [reply README.md]: One example of a README structure that you find very often on OSS projects:

https://gist.github.com/PurpleBooth/109311bb0361f32d87a2

File: src/test/java/ch/aaap/harvestclient/examples/UsersApiExample.java (hunk 2 above)
Comment (mnembrini) [reply src/test/java/ch/aaap/harvestclient/examples/UsersApiExample.java]: Ok makes sense, but in the rest of the tests I will put in the default constructor.

File: README.md (hunk 1 above)
Comment (mnembrini) [reply README.md]: I think a README should have all the information for a user of the library to start using it,
but a potential contributor to harvest-client can find more information in the wiki.
Usually there would be many more users than contributors, and they would not care about how to develop the library

For example I like the structure of https://square.github.io/retrofit/

**Pull request merged**